packages/proxy/lib/http/util/cookies.ts

```diff
@@ -0,0 +1,156 @@
+import _ from 'lodash'
+import { Cookie } from 'tough-cookie'
+import type Debug from 'debug'
+import { URL } from 'url'
+import { cors } from '@packages/network'
+import type { CookieJar } from '@packages/server/lib/cookie-jar'
+
+interface AutomationCookie {
+  domain: string
+  expiry: number | null
+  httpOnly: boolean
+  maxAge: number | null
+  name: string
+  path: string | null
+  sameSite: string
+  secure: boolean
+  value: string
+}
+
+interface RequestDetails {
+  url: string
+  isAUTFrame: boolean
+  needsCrossOriginHandling: boolean
+}
+
+// Sets SameSite context to match what it would be in the browser
+// see https://github.com/salesforce/tough-cookie#samesite-cookies
+export const getSameSiteContext = (autUrl: string | undefined, requestUrl: string, isAUTFrameRequest: boolean) => {
+  // if there's no AUT URL, it's a request for the first URL visited,
+  // so there would be no cross-origin concerns
+  // if the request origin matches the AUT origin, cookies can be handled
+  // in a strict fashion
+  if (!autUrl || cors.urlOriginsMatch(autUrl, requestUrl)) {
+    return 'strict'
+  }
+
+  // being an AUT frame request means it's from navigation, so the context is
+  // 'lax'. otherwise, 'none' indicates a non-navigation cross-origin request
+  return isAUTFrameRequest ? 'lax' : 'none'
+}
+
+const sameSiteNoneRe = /; +samesite=(?:'none'|"none"|none)/i
+
+export const parseCookie = (cookie) => {
+  const toughCookie = Cookie.parse(cookie)
+
+  if (!toughCookie) return
+
+  // fixes tough-cookie defaulting undefined/invalid SameSite to 'none'
+  // https://github.com/salesforce/tough-cookie/issues/191
+  const hasUnspecifiedSameSite = toughCookie.sameSite === 'none' && !sameSiteNoneRe.test(cookie)
+
+  if (hasUnspecifiedSameSite) {
+    toughCookie.sameSite = 'lax'
```

Comment:
Not all browsers default to 'lax'. Does that matter here?

https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#browser_compatibility

Reply:
That's true, but I think it's fine to err on the side of what the standard is, which is stricter and where browsers are heading anyway. I added a comment to explain this in the code too.

Reply:
Addressed in [`52b6874` (#22320)](https://github.com/cypress-io/cypress/pull/22320/commits/52b6874742ee589ed9c40b90792f3074bbf4d426).